lua/luasnip/nodes/snippet.lua

```diff
@@ -228,9 +254,12 @@ local function init_snippet_context(context, opts)
 	end
 	effective_context.trig_matcher = engine(effective_context.trigger)
 
-	effective_context.condition = context.condition
-		or opts.condition
-		or true_func
+	effective_context.resolve_expand_params =
+		generate_resolve_expand_params_func(
+			context.condition or opts.condition,
+			context.resolve_expand_params
+		)
+	effective_context.condition = context.condition or opts.condition
```

Comment:
Technically, I don't think we even have to store `condition` anymore, it's only used in `resolve_expand_params`, and we can put it in there via `generate_resolve_expand_params_func`

Reply:
(Oh btw, "gen_expand_params" would also be a good name, wdyt?)

Reply:
yeah, i will fix it later